src/engine/authentication.md

```diff
@@ -35,10 +35,24 @@ The HMAC algorithm implies that several CLs will be able to use the same key, an
 
 The `EL` and `CL` clients **MUST** accept a cli/config parameter: `jwt-secret`, which designates a file containing the hex-encoded 256 bit secret key to be used for verifying/generating JWT tokens.
 
-If such a parameter is not given, the client **SHOULD** generate such a token, valid for the duration of the execution, and store the hex-encoded secret as a `jwt.hex` file on the filesystem.  This file can then be used to provision the counterpart client.
-
 If such a parameter _is_ given, but the file cannot be read, or does not contain a hex-encoded key of `256` bits, the client should treat this as an error: either abort the startup, or show error and continue without exposing the authenticated port.
 
+If such a parameter is not given, the client **MUST** attempt to read the secret from the default paths defined below, in the order they are listed.  If a secret is found, but the file cannot be read, does not contain a hex-encoded key of `256` bit, or is rejected by the other client, the client should continue searching. The CL **MAY** search other locations, such as default EL data directories.
+
+If no existing JWT is located, the client **MUST** generate such a token, valid for the duration of the execution, and store the hex-encoded secret on the filesystem.  The exact path is dependant on the host OS, see the default paths below.  If the client is unable to write the secret to an OS-specific path, it **MUST** fallback to writing to its own data directory.
+
+### Default JWT secret locations
+
+For Linux:
+* `$XDG_DATA_DIR/ethereum/engine/jwt.hex`
```

Comment:
```suggestion
* `$XDG_CONFIG_HOME/ethereum/engine/jwt.hex`
```
I believe this qualifies as configuration, not data, and thus should go into the config directory.  From XDK spec:
>  There is a single base directory relative to which user-specific data files should be written. This directory is defined by the environment variable $XDG_DATA_HOME.
> 
> There is a single base directory relative to which user-specific configuration files should be written. This directory is defined by the environment variable $XDG_CONFIG_HOME.

Reply:
I'm not opposed to `XDG_CACHE_DIR`.  Given that this just needs to be set at startup and exist until the user launches the other client, cache seems quite reasonable as it isn't a big deal if this gets pruned by some cache cleanup mechanism.  That directory is also explicitly excluded from almost all backup systems which is a nice bonus.

Reply:
Updated.